test/reporter-html.js

```diff
@@ -100,27 +68,60 @@ QUnit.module( "timing", {
 
 QUnit.test( "setup", function( assert ) {
 	assert.expect( 0 );
-	delayNextSetup = true;
+	this.delayNextSetup = true;
 });
 
 QUnit.test( "basics", function( assert ) {
-	assert.expect( 2 );
-	var previous = getPreviousTests( /^setup$/, /^timing$/ )[ 0 ],
-		runtime = previous.lastChild.previousSibling;
-	assert.ok( /(^| )runtime( |$)/.test( runtime.className ), "Runtime element exists" );
+	assert.expect( 1 );
+	var previous = this.getPreviousTest( assert ),
+		runtime = this.filterClass( previous.getElementsByTagName( "span" ) );
+
 	assert.ok( /^\d+ ms$/.test( runtime.innerHTML ), "Runtime reported in ms" );
 });
 
 QUnit.test( "values", function( assert ) {
 	assert.expect( 2 );
-	var basics = getPreviousTests( /^setup$/, /^timing$/ )[ 0 ],
-		slow = getPreviousTests( /^basics$/, /^timing$/ )[ 0 ];
-	assert.ok( parseInt( basics.lastChild.previousSibling.innerHTML, 10 ) < 100,
+	var basics = this.getPreviousTest( assert ),
+		setup = basics.previousSibling;
+
+	basics = this.filterClass( basics.getElementsByTagName( "span" ) );
+	setup = this.filterClass( setup.getElementsByTagName( "span" ) );
+
+	assert.ok( parseInt( basics.innerHTML, 10 ) < 100,
 		"Fast runtime for trivial test"
 	);
-	assert.ok( parseInt( slow.lastChild.previousSibling.innerHTML, 10 ) > 100,
+	assert.ok( parseInt( setup.innerHTML, 10 ) > 100,
 		"Runtime includes beforeEach"
 	);
 });
 
-})();
+QUnit.module( "source" );
+
+QUnit.test( "setup", function( assert ) {
+	assert.expect( 0 );
+});
+
+QUnit.test( "logs location", function( assert ) {
+	var source,
+		previous = document.getElementById( "qunit-test-output-" + assert.test.testId  )
+			.previousSibling;
+
+	source = previous.lastChild;
+
+	if ( !Error().stack ) {
+		assert.equal(
+			/(^| )qunit-source( |$)/.test( source.className ),
+			false,
+			"Don't add source information on non-supported environments"
+		);
+		return;
+	}
+
+	assert.ok( /(^| )qunit-source( |$)/.test( source.className ), "Source element exists" );
+	assert.equal( source.firstChild.innerHTML, "Source: " );
+
+	// test/reporter-html.js is a direct reference to this test file
+	assert.ok( /\/test\/reporter-html\.js\:\d+/.test( source.innerHTML ),
+		"Source references to the current file and line number"
+	);
+});
```

Comment:
@gauravmittal1995, here are the tests for the source information that I was talking about.